packages/fxa-payments-server/src/components/PaymentErrorView/index.test.tsx

```diff
@@ -46,4 +58,49 @@ describe('PaymentErrorView test with l10n', () => {
 
     expect(onRetry).toHaveBeenCalled();
   });
+
+  it('renders as expected for a returning PayPal customer', () => {
+    const { queryByTestId, queryByAltText } = render(
+      <PaymentErrorView
+        onRetry={() => {}}
+        error={{ code: 'returning_paypal_customer_error' }}
+      />
+    );
+    const spinner = queryByAltText('error icon');
+    expect(spinner).toBeInTheDocument();
+
+    const mainBlock = queryByTestId('payment-error');
+    expect(mainBlock).toBeInTheDocument();
+
+    const expected =
+      'Sorry. Currently, you can only sign up for one subscription at a time. Please check back soon.';
+    const actual = getLocalizedMessage(
+      bundle,
+      'returning-paypal-customer-error',
+      {}
+    );
+    expect(actual).toEqual(expected);
+
+    const manageSubscriptionButton = queryByTestId('manage-subscription-link');
+    expect(manageSubscriptionButton).toBeInTheDocument();
+
+    const termsAndPrivacy = queryByTestId('terms');
+    expect(termsAndPrivacy).toBeInTheDocument();
+  });
+
+  it('does not call passed in onRetry function when "Manage my subscription" button clicked', async () => {
+    const onRetry = jest.fn();
+    const { getByTestId } = render(
+      <PaymentErrorView
+        onRetry={onRetry}
+        error={{ code: 'returning_paypal_customer_error' }}
+      />
+    );
+
+    await act(async () => {
+      fireEvent.click(getByTestId('manage-subscription-link'));
+    });
+
+    expect(onRetry).not.toHaveBeenCalled();
```

Comment:
This assertion that `onRetry` isn't called here is suboptimal. What I'd really like to do is assert that a different `onClick` handler was called; i.e. that `history.push("/subscriptions")` happened. Is there a better assertion I can use here?

Reply:
I think that checking the history would be better also. I see that you are mocking useHistory above. According to [this stack overflow answer](https://stackoverflow.com/a/59451956) You should be able to check the mocked history in the test.

Reply:
Good suggestion thanks; I found this at the time, but for whatever reason, I couldn't get it to work. Now it works though. Yay a good night's sleep?